gnucash/gnome/gnc-plugin-page-invoice.c

```diff
@@ -421,6 +467,30 @@ gnc_plugin_page_invoice_update_menus (GncPluginPage *page, gboolean is_posted, g
                                "sensitive", can_unpost);
     gnc_plugin_update_actions (action_group, invoice_book_readwrite_actions,
                                "sensitive", !is_readonly);
+
+
+    for (i = 0; i < parametrized_n_actions; i++)
+    {
+        const gchar *action_name = parametrized_actions[i];
+        // need action by name, check whether in plugin page actions
+        for (j = 0; j < gnc_plugin_page_invoice_n_actions; j++)
+        {
+            action = gnc_plugin_page_invoice_actions[j];
+            if (action_name == action.name)
```

Comment:
Minor performance consideration: you are iterating more than is needed. If the correct action is found, break out of the inner loop instead of continuing to scan the remaining actions.

More optimization could be achieved if you only loop over the plugin page actions once. That would require another approach though. As we are deeply in gui land here (using gtk and gtkactions) one way to achieve this would set a custom property on your parameterized actions (they are gobjects, so you can [add arbitrary data](https://developer.gnome.org/gobject/stable/gobject-The-Base-Object-Type.html#g-object-set-dataurl)). That custom property could be a gint with value 0 for a normal action or 1 for a parameterized option. I'm deliberately suggesting an int as that is one of the easiest to set/get as object data using `GINT_TO_POINTER` and `GPOINTER_TO_INT`
You can then loop over the plugin page's action, test the custom property and if it's 1, do the parameter substitution dance. For readability though it's probably best to use an enum and give the numbers 0 and 1 a clear enum name.

Reply:
True, I was planning to do that, but forgot to put it in.